app/components/ContextMenu/index.tsx

```diff
@@ -96,16 +99,21 @@ const ContextMenu: React.FC<Props> = ({
     t,
   ]);
 
-  // Perf win – don't render anything until the menu has been opened
+  // Perf win – render an empty menu until the menu has been opened
   if (!rest.visible && !previousVisible) {
-    return null;
+    return <Menu {...rest} />;
```

Comment:
I'd like to keep this perf gain of rendering nothing here, even rendering lots of "empty" menus adds up with all the hooks etc that are contained within.

Why was this needed? Lets think of another way

Reply:
Initially, the menu was rendered as a `modal` dialog on top of a popover.
In this setup, anytime the menu is opened, the popover was closed which ultimately closed the menu as well. By trial & error, I found that rendering an empty menu did not trigger the popover `hide` method.

Per your suggestion, to keep rendering nothing until the menu is opened, the menu is now rendered as a non-modal dialog which fixes the issue.